lib/statsig/user.ex

```diff
@@ -0,0 +1,60 @@
+defmodule Statsig.User do
+  @derive {Jason.Encoder, only: [
+    :userID,
+    :customIDs,
+    :email,
+    :ip,
+    :userAgent,
+    :country,
+    :locale,
+    :appVersion,
+    :custom,
+    :statsigEnvironment
+  ]}
+
+  @type custom_value :: String.t() | number() | boolean() | nil
+  @type custom_attributes :: %{optional(String.t()) => custom_value()}
+
+  @type t :: %__MODULE__{
+    userID: String.t() | nil,
+    customIDs: %{optional(String.t()) => String.t()} | nil,
+    email: String.t() | nil,
+    ip: String.t() | nil,
+    userAgent: String.t() | nil,
+    country: String.t() | nil,
+    locale: String.t() | nil,
+    appVersion: String.t() | nil,
+    custom: custom_attributes() | nil,
+    privateAttributes: custom_attributes() | nil,
+    statsigEnvironment: map() | nil
+  }
+
+  defstruct userID: nil,
+            customIDs: nil,
+            email: nil,
+            ip: nil,
+            userAgent: nil,
+            country: nil,
+            locale: nil,
+            appVersion: nil,
+            custom: nil,
+            privateAttributes: nil,
+            statsigEnvironment: nil
+
+  def new(params \\ %{}) do
```

Comment:
it's more idiomatic for the params to be a keyword list and not a map. Where's the map coming from? If it's coming from a web API, a `parse` or `cast` function is more appropriate. We can even make that completely automatic and safe with a small macro.